panther_lights/test/test_led_animation.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <yaml-cpp/yaml.h>
+
+#include <rclcpp/time.hpp>
+
+#include <panther_lights/led_animations_queue.hpp>
+#include <panther_lights/led_segment.hpp>
+
+class TestLEDAnimation : public testing::Test
+{
+public:
+  TestLEDAnimation();
+  ~TestLEDAnimation() {}
+
+  void SetSegmentAnimations();
+
+protected:
+  std::shared_ptr<panther_lights::LEDAnimation> led_anim_;
+  std::unordered_map<std::string, std::shared_ptr<panther_lights::LEDSegment>> segments_;
+};
+
+TestLEDAnimation::TestLEDAnimation()
+{
+  auto segment_1_desc = YAML::Load("{channel: 1, led_range: 0-10}");
+  auto segment_2_desc = YAML::Load("{channel: 2, led_range: 0-10}");
+  segments_.emplace(
+    "segment_1", std::make_shared<panther_lights::LEDSegment>(segment_1_desc, 50.0));
+  segments_.emplace(
+    "segment_2", std::make_shared<panther_lights::LEDSegment>(segment_2_desc, 50.0));
+
+  panther_lights::AnimationDescription anim_desc;
+  anim_desc.segments = {"segment_1", "segment_2"};
```

Comment:
The `segment_1` and `segment_2` are phrases commonly occurring in this test file. Would be useful to make global const vars for them to prevent accidental typo in further development.